xmppserver/src/main/java/org/jivesoftware/openfire/spi/ConnectionListener.java

```diff
@@ -1026,4 +1027,31 @@ public String toString()
                 '}';
     }
 
+    public void setStrictCertificateValidation(boolean strictCertificateValidation) {
+        final boolean oldValue = getStrictCertificateValidation();
+
+        // Always set the property explicitly even if it appears the equal to the old value (the old value might be a fallback value).
+        JiveGlobals.setProperty( type.getPrefix() + "certificate.strict-validation", Boolean.toString( strictCertificateValidation ) );
+
+        if ( oldValue == strictCertificateValidation ) {
+            Log.debug( "Ignoring strict certificate validation configuration change request (to '{}'): listener already in this state.", strictCertificateValidation );
+            return;
+        }
+
+        Log.debug( "Changing strict certificate validation configuration from '{}' to '{}'.", oldValue, strictCertificateValidation );
+        restart();
+    }
+
+    private boolean getStrictCertificateValidation() {
```

Comment:
Please add Javadoc and \*twitch\* move it above the `toString()` method.

Reply:
done